bagua/torch_api/algorithms/async_model_average.py

```diff
@@ -229,35 +229,56 @@ def _run_async_loop(self, bagua_module: BaguaModule):
             comm_step += 1
             time.sleep(self.sync_interval_ms / 1000)
 
-    def abort(self, bagua_module: BaguaModule):
+    def abort(self, bagua_ddp: Union[BaguaDistributedDataParallel]):
         """
-        Stop background asynchronous communications. Should be called after training.
+        Stop background asynchronous communications. Should be called after
+        training.
 
         Args:
-            bagua_module: A PyTorch module initialized by
-                :meth:`~bagua.torch_api.distributed.BaguaModule.with_bagua` method.
+            bagua_ddp (Union[BaguaDistributedDataParallel]): Bagua distributed data parallel module.
         """
 
+        if type(bagua_ddp) is BaguaDistributedDataParallel:
+            pass
+        elif hasattr(bagua_ddp, "inner"):
+            # Is bagua.torch_api.data_parallel.DistributedDataParallel
+            bagua_ddp = bagua_ddp.inner
+        elif hasattr(bagua_ddp, "bagua_ddp"):
+            # Is bagua.torch_api.distributed.BaguaModule
+            bagua_ddp = bagua_ddp.bagua_ddp
+        else:
+            raise Exception("unexcept input bagua_ddp={}".format(type(bagua_ddp)))
+
         if self.scheduled:
             barrier(comm=self.process_group.get_global_communicator())
             self.abort_event.set()
             self.future.result()  # pytype: disable=attribute-error
             self.scheduled = False
             logging.debug("Process {} async communication aborted.".format(get_rank()))
 
-    def resume(self, bagua_module: BaguaModule):
+    def resume(self, bagua_ddp: Union[BaguaDistributedDataParallel]):
         """
         Resume aborted background asynchronous communications (see :meth:`abort`). Should be called before training.
 
         Args:
-            bagua_module: A PyTorch module initialized by
-                :meth:`~bagua.torch_api.distributed.BaguaModule.with_bagua` method.
+            bagua_ddp (Union[BaguaDistributedDataParallel]): Bagua distributed data parallel module.
         """
 
+        if type(bagua_ddp) is BaguaDistributedDataParallel:
+            pass
+        elif hasattr(bagua_ddp, "inner"):
+            # Is bagua.torch_api.data_parallel.DistributedDataParallel
+            bagua_ddp = bagua_ddp.inner
+        elif hasattr(bagua_ddp, "bagua_ddp"):
+            # Is bagua.torch_api.distributed.BaguaModule
+            bagua_ddp = bagua_ddp.bagua_ddp
+        else:
+            raise Exception("unexcept input bagua_ddp={}".format(type(bagua_ddp)))
```

Comment:
unexpect? this error message is hard to understand. please refine

Reply:
DONE